testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java

```diff
@@ -542,6 +548,102 @@ public void testOrcTableCreatedInTrino()
         onTrino().executeQuery("DROP TABLE orc_table_created_in_trino");
     }
 
+    @Test(dataProvider = "storageFormatsWithConfiguration", groups = STORAGE_FORMATS_DETAILED)
+    public void testInsertAllSupportedDataTypesWithTrino(StorageFormat storageFormat)
+    {
+        // only admin user is allowed to change session properties
+        setAdminRole();
+        setSessionProperties(storageFormat);
+
+        String tableName = "storage_formats_compatibility_data_types_" + storageFormat.getName().toLowerCase(ENGLISH);
+
+        onTrino().executeQuery(format("DROP TABLE IF EXISTS %s", tableName));
+
+        final boolean isAvroStorageFormat = "AVRO".equals(storageFormat.name);
+        final boolean isParquetStorageFormat = "PARQUET".equals(storageFormat.name);
+        StorageFormatsHiveCompatibilityColumnData[] storageFormatsHiveCompatibilityTableData = new StorageFormatsHiveCompatibilityColumnData[] {
+                new StorageFormatsHiveCompatibilityColumnData("c_boolean", "BOOLEAN", "true", true),
+                new StorageFormatsHiveCompatibilityColumnData("c_tinyint", "TINYINT", "127", 127),
+                new StorageFormatsHiveCompatibilityColumnData("c_smallint", "SMALLINT", "32767", 32767),
+                new StorageFormatsHiveCompatibilityColumnData("c_int", "INTEGER", "2147483647", 2147483647),
+                new StorageFormatsHiveCompatibilityColumnData("c_bigint", "BIGINT", "9223372036854775807", 9223372036854775807L),
+                new StorageFormatsHiveCompatibilityColumnData("c_real", "REAL", "123.345", 123.345d),
+                new StorageFormatsHiveCompatibilityColumnData("c_double", "DOUBLE", "234.567", 234.567d),
+                new StorageFormatsHiveCompatibilityColumnData("c_decimal", "DECIMAL(10,0)", "346", new BigDecimal("346")),
+                new StorageFormatsHiveCompatibilityColumnData("c_decimal_w_params", "DECIMAL(10,5)", "345.67800", new BigDecimal("345.678")),
+                new StorageFormatsHiveCompatibilityColumnData("c_char", "CHAR(10)", "'ala ma    '", "ala ma    "),
+                new StorageFormatsHiveCompatibilityColumnData("c_varchar", "VARCHAR(10)", "'ala ma kot'", "ala ma kot"),
+                new StorageFormatsHiveCompatibilityColumnData("c_string", "VARCHAR", "'ala ma kota'", "ala ma kota"),
+                new StorageFormatsHiveCompatibilityColumnData("c_binary", "VARBINARY", "X'62696e61727920636f6e74656e74'", "binary content".getBytes(StandardCharsets.UTF_8)),
+                new StorageFormatsHiveCompatibilityColumnData("c_date", "DATE", "DATE '2015-05-10'", Date.valueOf(LocalDate.of(2015, 5, 10))),
```

Comment:
nit: by convention we use lowercase for identifiers, including Trino type names (`date`, `array(integer)`)
and uppercase for literal constructs (`DATE '2015-05-10'`, `ARRAY[1, 2, 3]`)